source/GameInterface/Services/SiegeEngineConstructionProgress/SiegeEngineConstructionProgressRegistry.cs

```diff
@@ -0,0 +1,22 @@
+﻿using GameInterface.Services.Registry;
+using System;
+using static TaleWorlds.CampaignSystem.Siege.SiegeEvent;
+
+namespace GameInterface.Services.SiegeEngines
+{
+    internal class SiegeEngineConstructionProgressRegistry : RegistryBase<SiegeEngineConstructionProgress>
+    {
+        public SiegeEngineConstructionProgressRegistry(IRegistryCollection collection) : base(collection)
+        {
+        }
+
+        public override void RegisterAll()
+        {
+        }
+
+        protected override string GetNewId(SiegeEngineConstructionProgress obj)
+        {
+            return Guid.NewGuid().ToString();
```

Comment:
Same thing as previous comment

Reply:
It appears that neither `SiegeEngineConstruction` or `SiegeEnginesContainer` can be used as type arg for `FindNextUniqueStringId` as they do not inherit `MBObjectBase`, should I do it like this instead?:
```
        private const string SiegeEngineConstructionProgressIdPrefix = "CoopSiegeEngineConstructionProgress";
        private static int InstanceCounter = 0;

        protected override string GetNewId(SiegeEngineConstructionProgress obj)
        {
            return $"{IdPrefix}_{Interlocked.Increment(ref InstanceCounter)}";
        }
```

Reply:
Ah okey, yes then that is what you will need 👍

Reply:
Added on both